pkgs/applications/networking/firewalld/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, docbook_xml_dtd_42
+, docbook_xsl
+, glib
+, gobject-introspection
+, gtk3
+, intltool
+, libnotify
+, libxml2
+, libxslt
+, networkmanager-applet
+, pkg-config
+, python3
+, wrapGAppsHook
+, withGui ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "firewalld";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "firewalld";
+    repo = "firewalld";
+    rev = "v${version}";
+    sha256 = "sha256-w8TbovIhOhJAUZWbKdBd/+db8Hro/ttlxWZDcrCXX4Q=";
+  };
+
+  postPatch = ''
+    for file in doc/xml/transform-{html,man}.xsl.in; do
+      substituteInPlace $file \
+        --replace "http://docbook.sourceforge.net/release/xsl/current" "${docbook_xsl}/xml/xsl/docbook"
```

Comment:
This should be picked up automatically from the XML catalog. Maybe `strictDeps` break the `findXMLCatalogs` setup hook?

Reply:
It does not even when strictDeps is disabled

```
...
checking for XML catalog (/nix/store/38yqc26w1vggk6mj4h3ra2n6nwz116ff-docbook-xsl-nons-1.79.2/xml/xsl/docbook/catalog.xml)... found
checking for xmlcatalog... /nix/store/rdrsw346kxmgww2w1wwrqry7l3v61wmf-libxml2-2.9.13-bin/bin/xmlcatalog
checking for DocBook XSL Stylesheets in XML catalog... found
...
I/O error : Attempt to load network entity http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl
warning: failed to load external entity "http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"
compilation error: file transform-man.xsl line 7 element import
xsl:import : unable to load http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl
...
```

Reply:
Looks like it is another instance of [macros](https://github.com/firewalld/firewalld/blob/79efc2535d13d74996cb2058c4873a3429110bbe/m4/jh_path_xml_catalog.m4#L44) not supporting `XML_CATALOG_FILES` environment variable. You can fix it using https://github.com/NixOS/nixpkgs/blob/aae4fc3e877a1eeb253cae1ff619cde20103a62d/pkgs/development/libraries/flatpak/respect-xml-catalog-files-var.patch or https://github.com/NixOS/nixpkgs/blob/aae4fc3e877a1eeb253cae1ff619cde20103a62d/pkgs/os-specific/linux/shadow/respect-xml-catalog-files-var.patch.